src/main/java/org/spongepowered/common/data/provider/DataProviderRegistrator.java

```diff
@@ -338,6 +338,9 @@ public R supports(final Function<H, Boolean> supports) {
 
                 @Override
                 protected Optional<E> getFrom(final H dataHolder) {
+                    if (registration.get == null) {
```

Comment:
Is there any time where this is actually going to be `null`? I can't imagine there is a use case for a write-only key within Sponge.

Reply:
In fact, it is, but who knows, maybe someday it will be. Given that the `get` field is declared with `@Nullable` annotation, I think the check should be.